src/installer.ts

```diff
@@ -0,0 +1,159 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import mkdirp = require('mkdirp');
+import {platform} from 'os';
+import { Errorable } from './errorable';
+import { Shell } from './shell';
+import * as child_process from 'child_process';
+import * as extension from './extension';
+import * as config from './config';
+import * as kamelCli from './kamel';
+
+const download = require('download-tarball');
+
+export const kamel = 'kamel';
+export const kamel_windows = 'kamel.exe';
+
+export function checkKamelCLIVersion() : Promise<string> {
+	return new Promise<string>( async (resolve, reject) => {
+        let kamelLocal = kamelCli.create();
+        await kamelLocal.invoke('version')
+            .then( (rtnValue) => {
+                const strArray = rtnValue.split(' ');
+                const version = strArray[strArray.length - 1].trim();
+                console.log(`Apache Camel K CLI (kamel) version returned: ${version}`);
+                resolve(version);
+                return;
+            }).catch ( (error) => {
+                console.log(`Apache Camel K CLI (kamel) unavailable: ${error}`);
+				reject(new Error(error));
+				return;
+        });
+    });
+}
+
+export function checkKubectlCLIVersion() : Promise<string> {
+	return new Promise<string>( (resolve, reject) => {
+        checkIfCLIAvailable('kubectl version')
+            .then( (rtnValue) => {
+                const strArray = rtnValue.split(' ');
+                strArray.forEach(element => {
+                    if (element.toLowerCase().startsWith('gitversion')) {
+                        const version = element.substring(element.indexOf('\"') + 1, element.lastIndexOf('\"'));
+                        console.log(`Kubernetes CLI (kubectl) version returned: ${version}`);
+                        resolve(version);
+                        return;
+                    }
+                });
+                reject (new Error('No kubectl version found'));
+                return;
+            }).catch ( (error) => {
+                console.log(`Kubernetes CLI (kubectl)  unavailable: ${error}`);
+				reject(new Error(error));
+				return;
+        });
+    });
+}
+
+export function checkMinikubeCLIVersion() : Promise<string> {
+	return new Promise<string>( (resolve, reject) => {
+        checkIfCLIAvailable('minikube version')
+            .then( (rtnValue) => {
+                const strArray = rtnValue.split(' ');
+                const version = strArray[strArray.length - 1].trim();
+                console.log(`Minikube CLI (minikube) version returned: ${version}`);
+                resolve(version);
+                return;
+            }).catch ( (error) => {
+                console.log(`Minikube CLI (minikube)  unavailable: ${error}`);
+				reject(new Error(error));
```

Comment:
Camel K can be installed on Kubernetes or OpenShift. it doesn't require Minikube. They are documenting 4 ways to install it: Minikube Minishift, Google Kubernetes Engine, OpenShift https://github.com/apache/camel-k#installation.

I think we can't have an error if minikube is not used.